### ISSUE TITLE
ArgumentNullException on controller discovery at design-time

### ISSUE THREAD

**Issue by natemcmaster** [aspnet/Mvc#4479]
Source issue: https://github.com/aspnet/EntityFramework/issues/5065

`DefaultAssemblyPartDiscoveryProvider.DiscoverAssemblyParts` throws if entryAssemblyName is null. This is condition happens when using a tool like dotnet-ef to discover user services.

Call stack:
```
 System.ArgumentNullException: Value cannot be null.
Parameter name: assemblyName
   at System.Reflection.AssemblyName..ctor(String assemblyName)
   at Microsoft.AspNetCore.Mvc.Internal.DefaultAssemblyPartDiscoveryProvider.DiscoverAssemblyParts(String entryPointAssemblyName)
   at Microsoft.Extensions.DependencyInjection.MvcCoreServiceCollectionExtensions.GetApplicationPartManager(IServiceCollection services)
```

**Comment by natemcmaster** [comment]
Triage: this appears to a blocking issue for using dotnet-ef on a web project.

**Comment by javiercn** [comment]
Hi, the issue is in https://github.com/aspnet/EntityFramework/blob/b5b9d2b3303093f7014248174b71d354a684e738/src/Microsoft.EntityFrameworkCore.Commands/Design/Internal/StartupInvoker.cs#L118.

MVC expects the hosting environment to have a valid application name. If you are adding your own hosting environment, you should initialize it there. In runtime scenarios the ApplicationName will never be null by default

We have an issue to track this and change this mechanism for a more resilient alternative in the future, but in any case, you should always initialize your hosting environment properly. https://github.com/aspnet/Mvc/issues/4353

**Comment by javiercn** [comment]
Closed as duplicate, tracking on #4353
